chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -2127,14 +2133,92 @@ impl PeerManagerActor {
+            if let Some(vc) = &state.config.validator {
+                let my_account_id = vc.signer.validator_id();
+                let my_public_key = vc.signer.public_key();
+                // TODO(gprusak): STUN servers should be queried periocally by a daemon
+                // so that the my_peers list is always resolved.
+                // Note that currently we will broadcast an empty list.
+                // It won't help us to connect the the validator BUT it
+                // will indicate that a validator is misconfigured, which
+                // is could be useful for debugging. Consider keeping this
+                // behavior for situations when the IPs are not known.
+                let my_peers = match &vc.endpoints {
+                    config::ValidatorEndpoints::TrustedStunServers(_) => vec![],
+                    config::ValidatorEndpoints::PublicAddrs(addrs) => {
+                        addrs.iter().cloned().map(|addr| PeerAddr { addr, peer_id: None }).collect()
+                    }
+                };
+                let my_data = info.tier1_accounts.iter().filter_map(|((epoch_id,account_id),key)| {
+                    if account_id != my_account_id{
+                        return None;
+                    }
+                    if key != &my_public_key {
+                        warn!(target: "network", "node's account_id found in TIER1 accounts, but the public keys do not match");
+                        return None;
+                    }
+                    Some(AccountData {
+                        epoch_id: epoch_id.clone(),
+                        account_id: my_account_id.clone(),
+                        timestamp: now,
+                        peers: my_peers.clone(),
+                    }.sign(vc.signer.as_ref()).unwrap())
+                }).collect();
+                // Insert node's own AccountData should never fail.
+                // We ignore the new data, because we trigger inserting
+                if let (_, Some(err)) = state.accounts_data.clone().insert(my_data).await {
+                    panic!("inserting node's own AccountData to self.state.accounts_data: {err}");
+                }
+            }
+            // The set of tier1 accounts has changed.
+            // We might miss some data, so we start a full sync with the connected peers.
+            // TODO(gprusak): add a daemon which does a periodic full sync in case some messages
+            // are lost (at a frequency which makes the additional network load negligible).
+            state
+                .clone()
+                .broadcast_message(SendMessage {
+                    message: PeerMessage::SyncAccountsData(SyncAccountsData {
+                        incremental: false,
+                        requesting_full_sync: true,
+                        accounts_data: state.accounts_data.dump(),
+                    }),
+                    context: Span::current().context(),
```

Comment:
We don't have any Span here. Prehaps we should though? "how much wall-clock/cpu time a broadcast takes" feels like an interesting data.

Reply:
added a span within the broadcast_message() call